langserver/diagnostics/diagnostics.go

```diff
@@ -3,81 +3,59 @@ package diagnostics
 import (
 	"context"
 	"log"
+	"path/filepath"
 	"sync"
 
 	"github.com/creachadair/jrpc2"
-	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/terraform-ls/internal/filesystem"
```

Comment:
I guess it's a nitpick, but we should probably extract the `uri*` stuff into its own dedicated `internal/uri` package, it feels awkward to be importing the whole `filesystem` just because of that function.

Reply:
I can do that in this PR, I think URIs (within the context of this project) is somewhat LSP specific, so could it move to `ilsp`?